Design/DesignDoc.md

```diff
@@ -0,0 +1,220 @@
+
+U9. As a user, I want to be able to view the remaining balance in my budget after my expenses for specified time period. 
+
+
+## 4. Project Scope
+
+_Clarify which parts of the problem you intend to solve. It helps reviewers know what questions to ask to make sure you are solving for what you say and stops discussions from getting sidetracked by aspects you do not intend to handle in your design._
+
+
+
+### 4.1. In Scope
+
+_Which parts of the problem defined in Sections 1 and 2 will you solve with this design? This should include the base functionality of your product. What pieces are required for your product to work?_
+
+* Adding, updating, and retrieving/viewing expenditure information
+* User specific data (multiple users allowed)
+* Retrieving expenditure information based on tags
+
+### 4.2. Out of Scope
+
+_Based on your problem description in Sections 1 and 2, are there any aspects you are not planning to solve? Do potential expansions or related problems occur to you that you want to explicitly say you are not worrying about now? Feel free to put anything here that you think your team can't accomplish in the unit, but would love to do with more time._
+
+* Displaying expenditure and budget information in a graphical representation 
+* Rolling over remaining budget information for the next term 
+* Calculating balance remaining in budget after expenses  
+
+
+
+
+# 5. Proposed Architecture Overview
+
+_Describe broadly how you are proposing to solve for the requirements you described in Section 2. This may include class diagram(s) showing what components you are planning to build. You should argue why this architecture (organization of components) is reasonable. That is, why it represents a good data flow and a good separation of concerns. Where applicable, argue why this architecture satisfies the stated requirements._
+
+This initial iteration will provide the minimum viable product (MVP) including adding, retrieving, and updating an expense information for the user role.
+
+We will use API Gateway and Lambda to create 6 endpoints (GetExpense, Add/UpdateExpense, GetTaggedExpense, GetBudget, Add/UpdateBudget, BudgetStatus) that will handle the creation, update, and retrieval of expense and budget information to satisfy our requirements.
+
+We will store expense and budget information in a table in DynamoDB.
+
+Amazon Elite Budget Service will also provide a web interface for users. A main page providing a variety of options to add, update and view expense and budget information.
+
+
+
+
+# 6. API
+
+## 6.1. Public Models
+
+_Define the data models your service will expose in its responses via your *`-Model`* package. These will be equivalent to the *`PlaylistModel`* and *`SongModel`* from the Unit 3 project._
+
+```
+// ExpenseModel
+
+String expenseId;
+String expenseName;
+String expenseAmount; 
+String tag;
+ZonedDateTime expenseDate;
+
+// BudgetModel
+
+String budgetId;
+String targetAmount; 
+Boolean status; 
```

Comment:
I might be mis-remembering, but I thought we talked about computing the "status" of a budget by querying for the expenses in a given period (vs setting a flag like this)?